From 09ddf24c0602256435dd4768ab5b3ccf1f6cd51c Mon Sep 17 00:00:00 2001
From: Holger Freyther <zecke@openmoko.org>
Date: Fri, 25 Jul 2008 23:06:05 +0100
Subject: [PATCH] [neo bluetooth] GTA01_GPIO_MODEM_RST != GTA02_GPIO_MODEM_RST
     The bluetooth enable and modem reset switched the order in
     gta02. Do not poke the bluetooth dongle when we want to reset
     the modem.

Signed-Off-By: Holger Freyther <zecke@openmoko.org>
---
 arch/arm/plat-s3c24xx/neo1973_pm_gsm.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-s3c24xx/neo1973_pm_gsm.c b/arch/arm/plat-s3c24xx/neo1973_pm_gsm.c
index cde3dd8..5ad942e 100644
--- a/arch/arm/plat-s3c24xx/neo1973_pm_gsm.c
+++ b/arch/arm/plat-s3c24xx/neo1973_pm_gsm.c
@@ -60,7 +60,9 @@ static ssize_t gsm_read(struct device *dev, struct device_attribute *attr,
 		if (s3c2410_gpio_getpin(GTA01_GPIO_MODEM_ON))
 			goto out_1;
 	} else if (!strcmp(attr->attr.name, "reset")) {
-		if (s3c2410_gpio_getpin(GTA01_GPIO_MODEM_RST))
+		if (machine_is_neo1973_gta01() && s3c2410_gpio_getpin(GTA01_GPIO_MODEM_RST))
+			goto out_1;
+		else if (machine_is_neo1973_gta02() && s3c2410_gpio_getpin(GTA02_GPIO_MODEM_RST))
 			goto out_1;
 	} else if (!strcmp(attr->attr.name, "download")) {
 		if (machine_is_neo1973_gta01()) {
@@ -135,7 +137,10 @@ static ssize_t gsm_write(struct device *dev, struct device_attribute *attr,
 			}
 		}
 	} else if (!strcmp(attr->attr.name, "reset")) {
-		s3c2410_gpio_setpin(GTA01_GPIO_MODEM_RST, on);
+		if (machine_is_neo1973_gta01())
+			s3c2410_gpio_setpin(GTA01_GPIO_MODEM_RST, on);
+		else if (machine_is_neo1973_gta02())
+			s3c2410_gpio_setpin(GTA02_GPIO_MODEM_RST, on);
 	} else if (!strcmp(attr->attr.name, "download")) {
 		if (machine_is_neo1973_gta01())
 			s3c2410_gpio_setpin(GTA01_GPIO_MODEM_DNLOAD, on);
-- 
1.5.6.3

